java/ql/src/Performance/StringReplaceAllWithNonRegex.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @id java/string-replace-all-with-non-regex
+ * @name Use of `String#replaceAll` with a first argument which is not a regular expression
+ * @description Using `String#replaceAll` is less performant than `String#replace` when the first
```

Comment:
As you know, I am not a developer, so I would be grateful if you could explain what the problem is with this being less performant. Does it take more time to execute?

Would something like be correct/acceptable?
(Sorry I hadn't hear of the adjective "performant" before. Discovered since that it is indeed a word in the engineering/computing area 😅 )

Using `String#replaceAll` instead of `String#replace` when the first argument is not a regular expression causes performance issues.

Feel free to ignore if you think that developers will understand what this query does with the description as-is.